tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -305,6 +305,106 @@ def test_torch_whisper(self):
         output = speech_recognizer([filename], chunk_length_s=5, batch_size=4)
         self.assertEqual(output, [{"text": " A man said to the universe, Sir, I exist."}])
 
+    @slow
+    @require_torch
+    def test_whisper_timestamp_prediction(self):
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
+        array = np.concatenate(
+            [ds[40]["audio"]["array"], ds[41]["audio"]["array"], ds[42]["audio"]["array"], ds[43]["audio"]["array"]]
+        )
+        pipe = pipeline(
+            model="openai/whisper-tiny",
+            return_timestamps=True,
+        )
+
+        output = pipe(ds[40]["audio"])
+        self.assertDictEqual(
+            output,
+            # TODO This is not great, we're missing the last segment.
+            {
+                "text": " A man said to the universe, Sir, I exist.",
+                "chunks": [{"text": " A man said to the universe,", "timestamp": (0.0, 2.48)}],
+            },
+        )
```

Comment:
This test is only decoding the sequence, and indeed there is a simple bug : we are not adding the last timestamp token (which is usually not consecutive) when decoding.

Reply:
(just fixed it, will push)